### PR TITLE
refactor: adjust project_api_key to new file structure

### PR DESCRIPTION
**Pull request by AgustinBettati**
## Description

Link to any related issue(s): [INTMDB-1387](https://jira.mongodb.org/browse/INTMDB-1387)

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.

### CODE REVIEW

File: internal/provider/provider_sdk2.go [hunk 1]
```diff
@@ -108,8 +109,8 @@ func getDataSourcesMap() map[string]*schema.Resource {
 		"mongodbatlas_api_keys":                          mongodbatlas.DataSourceAPIKeys(),
 		"mongodbatlas_access_list_api_key":               mongodbatlas.DataSourceAccessListAPIKey(),
 		"mongodbatlas_access_list_api_keys":              mongodbatlas.DataSourceAccessListAPIKeys(),
-		"mongodbatlas_project_api_key":                   mongodbatlas.DataSourceProjectAPIKey(),
-		"mongodbatlas_project_api_keys":                  mongodbatlas.DataSourceProjectAPIKeys(),
+		"mongodbatlas_project_api_key":                   projectapikey.DataSourceProjectAPIKey(),
```
Comment (gssbzn) [inline internal/provider/provider_sdk2.go:112]: [nit-non-blocking] given the decision on one package per resource and to follow the "don't stutter" general recommendation of go 
```suggestion
		"mongodbatlas_project_api_key":                   projectapikey.DataSource(),
```

File: mongodbatlas/data_source_org_id.go [hunk 2]
```diff
@@ -42,7 +43,7 @@ func dataSourceMongoDBAtlasOrgIDRead(ctx context.Context, d *schema.ResourceData
 	for idx, role := range apiKeyOrgList.APIKey.Roles {
 		if strings.HasPrefix(role.RoleName, "ORG_") {
 			if err := d.Set("org_id", apiKeyOrgList.APIKey.Roles[idx].OrgID); err != nil {
-				return diag.Errorf(ErrorProjectSetting, `org_id`, root.APIKey.ID, err)
+				return diag.Errorf(projectapikey.ErrorProjectSetting, `org_id`, root.APIKey.ID, err)
```
Comment (gssbzn) [inline mongodbatlas/data_source_org_id.go]: this error seems to be defined in the wrong package

**Review by gssbzn — APPROVED** [review]

File: mongodbatlas/data_source_org_id.go (hunk 2 above)
Comment (AgustinBettati) [reply mongodbatlas/data_source_org_id.go]: yeah this message was not accurate, adjusted and left a new constant to help with consistency in error messages

File: internal/provider/provider_sdk2.go (hunk 1 above)
Comment (AgustinBettati) [reply internal/provider/provider_sdk2.go:112]: this makes sense, will do a followup PR to adjust all of the resources that have already been adjusted to new structure.

Comment (lantoli) [reply internal/provider/provider_sdk2.go:112]: nice, thx

**Review by lantoli — APPROVED** [review]

**Comment by lantoli** [comment]
the failing test is the flaky test so feel free to merge even if that failing test

**Pull request merged**